raft/src/test/java/org/apache/kafka/raft/RaftClientTestContext.java

```diff
@@ -641,19 +641,43 @@ void assertSentDescribeQuorumResponse(
         long highWatermark,
         List<ReplicaState> voterStates,
         List<ReplicaState> observerStates
+    ) {
+        short apiVersion = (short) (kip853Rpc ? 2 : 1);
+        assertSentDescribeQuorumResponse(leaderId, leaderEpoch, highWatermark, voterStates, observerStates, apiVersion, Errors.NONE);
+    }
+
+    void assertSentDescribeQuorumResponse(
+        int leaderId,
+        int leaderEpoch,
+        long highWatermark,
+        List<ReplicaState> voterStates,
+        List<ReplicaState> observerStates,
+        short apiVersion,
+        Errors error
     ) {
         DescribeQuorumResponseData response = collectDescribeQuorumResponse();
 
         DescribeQuorumResponseData.PartitionData partitionData = new DescribeQuorumResponseData.PartitionData()
-            .setErrorCode(Errors.NONE.code())
+            .setErrorCode(error.code())
             .setLeaderId(leaderId)
             .setLeaderEpoch(leaderEpoch)
             .setHighWatermark(highWatermark)
             .setCurrentVoters(voterStates)
             .setObservers(observerStates);
 
-        // KAFKA-16953 will add support for including the node listeners in the node collection
-        DescribeQuorumResponseData.NodeCollection nodes = new DescribeQuorumResponseData.NodeCollection();
+        if (!error.equals(Errors.NONE)) {
+            partitionData.setErrorMessage(error.message());
+        }
+
+        DescribeQuorumResponseData.NodeCollection nodes = null;
+        if (apiVersion >= 2) {
+            nodes = new DescribeQuorumResponseData.NodeCollection(voterStates.size());
```

Comment:
I think the default is the empty list not the null value. For example, take a look at the implementation of `write` in `DescribeQuorumResponseData`:
```java
          if (_version >= 2) {
              _writable.writeUnsignedVarint(nodes.size() + 1);
              for (Node nodesElement : nodes) {
                  nodesElement.write(_writable, _cache, _version);
              }
          } else {
              if (!this.nodes.isEmpty()) {
                  throw new UnsupportedVersionException("Attempted to write a non-default nodes at version " + _version);
              }
          }
```

That code assumes that the value is not equals to `null`.